indico/web/templates/forms/category_picker_widget.html

```diff
@@ -0,0 +1,40 @@
+{% extends 'forms/base_widget.html' %}
+
+
+{% block html %}
+    <div class="i-form-field-fixed-width" data-tooltip-anchor>
+        <div id="category-title-{{ field.id }}" class="text-holder"></div>
+        <a class="i-button" id="category-picker-button-{{ field.id }}"
+           data-title='{% trans %}Choose category{% endtrans %}'>
+            {%- trans %}Choose category{% endtrans -%}
+        </a>
+        <input type="hidden" id="{{ field.id }}" name="{{ field.name }}" value="">
+    </div>
+{% endblock %}
+
+
+{% block javascript %}
+    <script>
+        (function() {
+            'use strict';
+
+            var $field = $('#{{ field.id }}');
+            var $categoryTitle = $('#category-title-{{ field.id }}');
+            var $dialogTrigger = $('#category-picker-button-{{ field.id }}');
+
+            $dialogTrigger.on('click', function(evt) {
+                evt.preventDefault();
+                $('<div>').categorypicker({
+                    openInDialog: true,
+                    onAction: function(category) {
+                        $categoryTitle.text(category.title);
+                        $field.val(category.id)
+                            .data('is-protected', category.is_protected)
+                            .data('category-title', category.title)
+                            .trigger('change');
```

Comment:
Singe we trigger a change for any other element from outside to use whatever it was returned, I would not only add `is-protected` and `category-title` but the whole category data.